Kernel/ProcessGroup.cpp

```diff
@@ -30,10 +30,12 @@ RefPtr<ProcessGroup> ProcessGroup::create(ProcessGroupID pgid)
 
 RefPtr<ProcessGroup> ProcessGroup::find_or_create(ProcessGroupID pgid)
 {
-    ScopedSpinLock lock(g_process_groups_lock);
+    {
+        ScopedSpinLock lock(g_process_groups_lock);
 
-    if (auto existing = from_pgid(pgid))
-        return existing.release_nonnull();
+        if (auto existing = from_pgid(pgid))
+            return existing;
+    }
 
     return create(pgid);
```

Comment:
Hold up, isn't this racy if two threads are calling `ProcessGroup::find_or_create()` at the same time? Both of them could end up in `create()`, creating two of the same `ProcessGroup` and putting both in the global list.

Reply:
True, reworked this to avoid the race.